domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsGreaterThanRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,447 @@
+  private val MIXED_NUMBER_123_1_OVER_3 =
+    createMixedNumber(isNegative = false, wholeNumber = 123, numerator = 1, denominator = 3)
+  private val NEGATIVE_MIXED_NUMBER_123_1_OVER_2 =
+    createMixedNumber(isNegative = true, wholeNumber = 123, numerator = 1, denominator = 2)
+  private val NEGATIVE_MIXED_NUMBER_123_1_OVER_3 =
+    createMixedNumber(isNegative = true, wholeNumber = 123, numerator = 1, denominator = 3)
+  private val STRING_VALUE = createString(value = "test")
+
+  @Test
+  fun testPositiveFractionalAnswer_positiveFractionalInput_exactSameValue_answerNotGreater() {
+    val inputs = mapOf("f" to FRACTION_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = FRACTION_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testPositiveFractionalAnswer_positiveFractionalInput_answerGreaterThan_answerGreater() {
+    val inputs = mapOf("f" to FRACTION_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = FRACTION_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testPositiveFractionalAnswer_positiveFractionalInput_answerSmallerThan_answerNotGreater() {
+    val inputs = mapOf("f" to FRACTION_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = FRACTION_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeFractionalAnswer_negativeFractionalInput_exactSameValue_answerNotGreater() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_FRACTION_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeFractionalAnswer_negativeFractionalInput_answerGreaterThan_answerGreater() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_1_OVER_2)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_FRACTION_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testNegativeFractionalAnswer_negativeFractionalInput_answerSmallerThan_answerNotGreater() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = NEGATIVE_FRACTION_1_OVER_2,
+      inputs = inputs
+    )
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testPositiveFractionalAnswer_negativeFractionalInput_answerGreaterThan_answerGreater() {
+    val inputs = mapOf("f" to NEGATIVE_FRACTION_1_OVER_3)
+
+    val matches = inputGreaterThanRuleClassifier.matches(
+      answer = FRACTION_1_OVER_3,
+      inputs = inputs
+    )
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testNegativeFractionalAnswer_PositiveFractionalInput_answerSmallerThan_answerNotGreater() {
```

Comment:
'p' in the Positive should be in small case.

Reply:
done